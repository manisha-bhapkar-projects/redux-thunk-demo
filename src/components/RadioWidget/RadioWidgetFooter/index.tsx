import { useContext } from 'react';
import { RadioWidgetContext, RadioStationObject } from '../index';


const RadioWidgetFooter = () => {
    const RADIO_CONTEXT = useContext(RadioWidgetContext);
    return (
        <div className="custom-card-footer">
            <div className="footer-content">
                {RADIO_CONTEXT.selectedStation ?
                    <span>CURRENTLY PLAYING</span> : ""
                }
                <h4>{RADIO_CONTEXT.selectedStation?.station_name}</h4>
            </div>
        </div>

    )
}
export default RadioWidgetFooter;