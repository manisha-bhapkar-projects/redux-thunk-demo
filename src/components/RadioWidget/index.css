.custom-card {
    width: 300px;
    margin: auto;
    border-radius: 35px;
    box-shadow: 0px 24px 24px -8px rgba(0, 0, 0, 0.4);
}

.custom-card-header {
    display: flex;
    background-color: #ecaa5b;
    padding: 20px 30px;
    border-top-left-radius: 35px;
    border-top-right-radius: 35px;
    display: flex;
    align-items: center;
    justify-content: space-between;
}

/* .custom-card-header img{
    width: 30px;
} */
.arrow-img {
    width: 16px;
}

.power-img {
    width: 25px;
}

/* .fm-point-show img{
    width: 45%;
} */
.custom-card-header h1 {
    color: #fff;
    margin-bottom: 0;
    font-size: 22px;
    text-transform: uppercase;
    font-weight: 600;
    margin-top: 0;
}

.custom-card-header img:nth-child(1) {
    width: 15px;
}

.custom-card-header img:nth-last-child(1) {
    width: 25px;
}

.custom-card-body {
    background-color: #30303b;
    padding: 0 30px;
    height: 260px;
    overflow-y: auto;
}

.custom-card-body {}

.custom-card-body::-webkit-scrollbar {
    width: 0;
    /* Remove scrollbar space */
    background: transparent;
    /* Optional: just make scrollbar invisible */
}

/* Optional: show position indicator in red */
.custom-card-body::-webkit-scrollbar-thumb {
    background: #FF0000;
}

.custom-card-body-content {
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid #a2abbd;
    padding-bottom: 15px;
    padding-top: 15px;
}

.custom-card-body-content:nth-last-child(1) {
    border-bottom: none;
}

.custom-card-body-content h2 {
    font-weight: 300;
    color: #a2abbd;
    font-size: 18px;
    margin-bottom: 0;
}

.custom-card-body-content a {
    display: flex;
}

.custom-card-body-content h3 {
    color: #a2abbd;
    font-size: 18px;
    font-weight: 600;
    margin-bottom: 0;
    /* margin-top: 0; */
}

.custom-card-footer {
    background-color: #25252f;
    border-top: 1px solid #515561;
    padding: 10px 0 10px 0;
    border-bottom-left-radius: 35px;
    border-bottom-right-radius: 35px;
    text-align: center;
    height: 80px;
}

.custom-card-footer span {
    color: #ecaa5b;
    font-size: 12px;
}

.custom-card-footer h4 {
    color: #a2abbd;
    font-size: 22px;
    font-weight: 400;
}

.fm-point-show {
    display: flex;
    width: 100%;
    justify-content: space-between;
    align-items: center;
    padding-top: 40px;
    transition: all 2s linear;
}

.fm-point-show a:nth-child(1) {
    text-align: left;
}

.fm-point-show a:nth-last-child(1) {
    text-align: right;
}

.fm-point-show img {
    width: 50%;
}

a {
    text-decoration: none;
}

a:hover {
    text-decoration: none;
}

.footer-content {
    height: 60px;
}

.btn-transpernt {
    background: transparent;
}

.stationbutton {
    background-color: #30303b;
    border: none;
    cursor: pointer;
}

.stationbutton:focus {
    outline: none;
}