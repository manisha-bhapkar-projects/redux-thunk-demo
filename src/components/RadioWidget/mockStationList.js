export default [
            {
                "station_id": 1,
                "station_name": "Putin FM",
                "station_frequency": "66,6"
            },
            {
                "station_id": 2,
                "station_name": "Dribbble FM",
                "station_frequency": "101,2"
            },
            {
                "station_id": 3,
                "station_name": "Doge FM",
                "station_frequency": "99,4"
            },
            {
                "station_id": 4,
                "station_name": "Ballads FM",
                "station_frequency": "87,1"
            },
            {
                "station_id": 5,
                "station_name": "Maximum FM",
                "station_frequency": "142,2"
            }
        ]