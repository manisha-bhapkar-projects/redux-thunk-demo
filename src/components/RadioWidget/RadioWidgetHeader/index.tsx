
import { SwitchIcon, BackArrowIcon } from '../../../components/Icon';

type RadioWidgetHeaderProps = {
    label?: string;
    handleBackButton?: Function;
    handleSwitchButton?: Function;
}

const RadioWidgetHeader = (props: RadioWidgetHeaderProps) => {
    return (<div className="custom-card-header">
        <BackArrowIcon onClick={props.handleBackButton} />
        <h1>{props.label ? props.label : 'Station'}</h1>
        <SwitchIcon onClick={props.handleSwitchButton} />
    </div>
    )
}
export default RadioWidgetHeader;