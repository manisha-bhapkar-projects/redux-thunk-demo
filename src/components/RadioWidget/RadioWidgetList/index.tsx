import { useContext } from 'react';
import { PlusIcon, MinusIcon } from '../../../components/Icon';
import PlaceholderImage from '../../../assets/placeholder.png';
import Icon from '../../Icon';
import { RadioWidgetContext } from '../index';
import LoaderComponent from '../../Loader/LoaderComponent'

type RadioWidgetListProps = {
    list: Array<any>,
    loading: boolean
    handleSelectStation: Function,
    handlePlusButton?: Function,
    handleMinusButton?: Function,
}

const RadioWidgetList = (props: RadioWidgetListProps) => {
    const RADIO_CONTEXT = useContext(RadioWidgetContext);

    return (
        <div className="custom-card-body">
            {
                props.loading ?
                    <div className="d-flex justify-content-center align-items-center w-100 h-100">
                        <LoaderComponent
                            type="Circles"
                            color="#ecaa5b"
                            height={50}
                            width={50}
                            timeout={5000}
                        />
                    </div>
                    :
                    props.list &&
                    props.list
                        .map((station: any) => {
                            return (
                                <>
                                    {station.station_id == RADIO_CONTEXT.selectedStation?.station_id ?
                                        <div className="fm-point-show" key={station.station_id}>
                                            <button className="stationbutton" >
                                                <MinusIcon onClick={props.handleMinusButton} />
                                            </button>
                                            <Icon icon={PlaceholderImage} />
                                            <button className="stationbutton" >
                                                <PlusIcon onClick={props.handlePlusButton} />
                                            </button>
                                        </div>
                                        : ""}
                                    <a className="custom-card-body-content"
                                        onClick={() => props.handleSelectStation(station)}>
                                        <h2>{station.station_name}</h2>
                                        <h3>{station.station_frequency}</h3>
                                    </a>
                                </>
                            )
                        })

            }

        </div>
    )
}
export default RadioWidgetList;