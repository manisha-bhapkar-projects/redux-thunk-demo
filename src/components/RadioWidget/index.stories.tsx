import { Meta, Story } from '@storybook/react'
import RadioWidget, { RadioWidgetProps } from '../RadioWidget'
import mockStationList from './mockStationList'
export default {
    title: 'Components/RadioWidget',
    component: RadioWidget,
    args: {
        stationList: mockStationList
    }
} as Meta

const Template: Story<RadioWidgetProps> = args => <RadioWidget  {...args} />

export const Basic = Template.bind({})
Basic.args = {}
