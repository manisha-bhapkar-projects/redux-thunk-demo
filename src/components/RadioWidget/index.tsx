import React from 'react';
import RadioWidgetList from './RadioWidgetList';
import RadioWidgetFooter from './RadioWidgetFooter';
import RadioWidgetHeader from './RadioWidgetHeader';
import './index.css';
export type RadioWidgetProps = {
    cardTitle?: string;
    stationList: Array<RadioStationObject>;
    loading: boolean
    getSelectedStation?: (RadioStationObject) => void;
    handlePlusButton?: () => void;
    handleMinusButton?: () => void;
    handleBackButton?: () => void;
    handleSwitchButton?: () => void;
}

export type RadioStationObject = {
    station_id: Number;
    station_name: string;
    station_frequency: String;
}
export const RadioWidgetContext = React.createContext<any>(0);

const RadioWidget = (props: RadioWidgetProps) => {
    const { Provider } = RadioWidgetContext;
    const [selectedStation, setselectedStation] = React.useState<RadioStationObject>();

    const handleStationSelect = (data: RadioStationObject) => {
        if (selectedStation?.station_id == data.station_id) {
            setselectedStation(undefined)
            props.getSelectedStation && props.getSelectedStation(undefined)
        } else {
            setselectedStation(data);
            props.getSelectedStation && props.getSelectedStation(data)
        }
    }

    return (
        <div className="custom-card">
            <Provider
                value={{
                    selectedStation: selectedStation
                }}
            >
                <RadioWidgetHeader label={props.cardTitle}
                    handleBackButton={props.handleBackButton}
                    handleSwitchButton={props.handleSwitchButton} />
                <RadioWidgetList
                    list={props.stationList}
                    loading={props.loading}
                    handleSelectStation={handleStationSelect}
                    handleMinusButton={props.handleMinusButton}
                    handlePlusButton={props.handlePlusButton}

                />
                <RadioWidgetFooter />
            </Provider>
        </div>
    );
}
export default RadioWidget;