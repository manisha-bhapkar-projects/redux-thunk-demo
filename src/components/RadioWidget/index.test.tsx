import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import RadioWidget from './index';
import mockStationList from './mockStationList'
const createTestProps = (props) => ({
    stationList: mockStationList,
    ...props
})
describe("Radio Widget component test suite", () => {
    it('it should render successfully', () => {
        let props = createTestProps({
            cardTitle: 'Custom Card Title',
            handlePlusButton: jest.fn(),
            handleMinusButton: jest.fn(),
            handleBackButton: jest.fn(),
            handleSwitchButton: jest.fn()
        })
        const comp = shallow(
            <RadioWidget {...props} />
        );
        expect(toJson(comp.container)).toMatchSnapshot();
    });
});