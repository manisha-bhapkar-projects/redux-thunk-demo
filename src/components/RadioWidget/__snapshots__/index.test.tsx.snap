// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Widget component test suite it should render successfully 1`] = `undefined`;
